geomesa-jupyter/src/main/scala/org/locationtech/geomesa/jupyter/Leaflet.scala

```diff
@@ -0,0 +1,166 @@
+         |  fillOpacity: '$fillOpacity'
+         |}
+       """.stripMargin
+  }
+
+  case class WMSLayer(layerName: String,
+                      style: String = "",
+                      filter: String = "INCLUDE",
+                      color: String = "#FF0000",
+                      geoserverURL: String = "/geoserver/wms",
+                      env: Map[String,String] = Map.empty[String,String],
+                      opacity: Double = 0.6,
+                      transparent: Boolean = true) extends GeoRenderable {
+    override def render: String =
+      s"""
+         | L.WMS.source('$geoserverURL?',
+         |   {
+         |      layers: '$layerName',
+         |      cql_filter: "$filter",
+         |      styles: '$style',
+         |      env: '${env.map { case (k,v) => Array(k,v).mkString(sep = "=")}.mkString(sep = ":")}',
+         |      transparent: '$transparent',
+         |      opacity: $opacity,
+         |      format: 'image/png',
+         |      version: '1.1.1'
+         |   }).getLayer('$layerName').addTo(map);
+         |
+       """.stripMargin
+  }
+  case class SimpleFeatureLayer(features: Seq[SimpleFeature], style: StyleOptions) extends GeoRenderable {
+    override def render: String =
+      s"""
+         |L.geoJson(${features.map(simpleFeatureToGeoJSON).mkString("[",",","]")},
+         |    { "onEachFeature": featurePopup, style: '$style.render' }
+         |).addTo(map);
+       """.stripMargin
+
+    private def simpleFeatureToGeoJSON(sf: SimpleFeature) = {
+      import scala.collection.JavaConversions._
+      s"""
+         |{
+         |    "type": "Feature",
+         |    "properties": {
+         |        ${sf.getType.getAttributeDescriptors.zip(sf.getAttributes).filter{case(a,b) => b != null}.map { case (d, a) => propToJson(d, a) }.mkString(sep =",\n")}
+         |    },
+         |    "geometry": {
+         |        "type": "${sf.getDefaultGeometry.asInstanceOf[Geometry].getGeometryType}",
+         |        "coordinates": ${processGeometry(sf.getDefaultGeometry.asInstanceOf[Geometry])}
+         |    }
+         |}
+       """.stripMargin
+    }
+
+    private def propToJson(ad: AttributeDescriptor, a: Object) =
+      if(a!= null) s""""${ad.getLocalName}": '${StringEscapeUtils.escapeJson(a.toString)}'"""
+      else s"${ad.getLocalName}: ''"
+
+    private def processGeometry(geom: Geometry) = geom.getGeometryType match {
+      case "Point"      => processPoint(geom.asInstanceOf[Point])
+      case "LineString" => processLinestring(geom.asInstanceOf[LineString])
+      case "Polygon"    => processPoly(geom.asInstanceOf[Polygon])
+    }
+
+    def processCoord(c: Coordinate) = s"[${c.x}, ${c.y}]"
+    def processPoint(p: Point) = s"${processCoord(p.getCoordinate)}"
+    def processLinestring(l: LineString) = s"[${l.getCoordinates.map { c =>processCoord(c)}.mkString(",")}]"
+    def processPoly(p: Polygon) = s"[${p.getCoordinates.map { c =>processCoord(c)}.mkString(",")}]"
```

Comment:
do we care about exterior ring vs interior holes in polygons? Not sure how `getCoordinates` handles those